securedrop/journalist_app/api2/events.py

```diff
@@ -54,6 +55,11 @@ def process(self, event: Event) -> EventResult:
             handler = {
                 EventType.ITEM_DELETED: self.handle_item_deleted,
                 EventType.REPLY_SENT: self.handle_reply_sent,
+                EventType.SOURCE_DELETED: self.handle_source_deleted,
+                EventType.SOURCE_CONVERSATION_DELETED: self.handle_source_conversation_deleted,
+                EventType.SOURCE_STARRED: self.handle_source_starred,
+                EventType.SOURCE_UNSTARRED: self.handle_source_unstarred,
+                EventType.SOURCE_SEEN: self.handle_source_seen,
```

Comment:
should this instead be `ITEM_SEEN`? since this is just for marking an item as seen?

Reply:
I wonder if in the case of `SEEN`, the client should submit events via batch, but then when we receive a successful response from a `SEEN` item event, that doesn't necessarily mean that we can mark the entire source as seen. The source seen state we may have to update via the `sources` payload?